Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -1064,22 +1130,24 @@ void MediaSource::sourceBufferDidChangeActiveState(SourceBuffer&, bool)
     regenerateActiveSourceBuffers();
 }
 
-bool MediaSource::attachToElement(HTMLMediaElement& element)
+bool MediaSource::attachToElement(WeakPtr<HTMLMediaElement>&& element)
```

Comment:
Maybe we should add two attachToElement, one with a HTMLMediaElement&, called by the main proxy, and one with a WeakPtr<>&& called by the worker proxy.

The second one would do nothing but set m_mediaElement, and would probably not even return a boolean.